src/renderer/components/data-settings/data-settings.js

```diff
@@ -335,79 +334,98 @@ export default Vue.extend({
     },
 
     importOpmlYouTubeSubscriptions: async function (data) {
-      let json
+      let xmlDom
+      const domParser = new DOMParser()
       try {
-        json = await opmlToJSON(data)
+        xmlDom = domParser.parseFromString(data, 'application/xml')
+
+        // https://developer.mozilla.org/en-US/docs/Web/API/DOMParser/parseFromString#error_handling
+        const errorNode = xmlDom.querySelector('parsererror')
+        if (errorNode) {
+          throw errorNode.textContent
+        }
       } catch (err) {
-        console.error(err)
-        console.error('error reading')
-        const message = this.$t('Settings.Data Settings.Invalid subscriptions file')
-        showToast(`${message}: ${err}`)
-      }
+        // try parsing with the html parser instead which is more lenient
+        try {
+          const htmlDom = domParser.parseFromString(data, 'text/html')
 
-      if (json !== undefined) {
-        let feedData = json.children[0].children
-        if (typeof feedData === 'undefined') {
-          if (json.title.includes('gPodder')) {
-            feedData = json.children
-          } else {
-            const message = this.$t('Settings.Data Settings.Invalid subscriptions file')
-            showToast(message)
-            return
-          }
+          xmlDom = htmlDom
+        } catch {
```

Comment:
Would you need to throw the errornode again to get to this catch block?

Reply:
Not sure, the docs don't mention what happens if it fails to parse HTML https://developer.mozilla.org/en-US/docs/Web/API/DOMParser/parseFromString#error_handling

The HTML parser is too lenient, for the broken file it produced this:
![broken_file](https://user-images.githubusercontent.com/48293849/198275517-cc080ba8-7d0b-4a05-b4f0-61c86255e422.jpg)

Maybe I should check the error message and only fall back to the HTML parser if the HTML entity error occurs?

Reply:
Okay, only doing it for a specific error won't work, as Firefox always outputs the same error if the XML is problematic. So checking the error, wouldn't work for the web build.

Reply:
Might be good to log the error with the xml and then try to do it with the html dom parser

Reply:
It'll log the error now before trying to parse it with the HTML parser.